### DISCUSSION TITLE
I've decided to open source my personal IoT framework

### DISCUSSION THREAD

**Discussion by wolfen351** [micropython/micropython#14051] [Show and tell]
Hi all

I'm proud to finally show the world my baby IoT platform. This is about a year's worth of development in micropython in my spare time. I've gone for a modular approach where you can essentially make whatever gadget you want by assembling modules like a touch screen and a temp sensor and a web server for example to make a touch temp sensor gadget.

Here is a link to the github - I'm still working on documenting all the modules and adding in thanks for all the great open source projects I've leaned on.  https://github.com/wolfen351/public-micropython-iot-platform 

I'd really appreciate feedback, and potentially others excited to help build this with me! 

I've made some fun projects - a ultrasonic garage door closer, a ac_remote control panel with a touch screen, a binary clock, many temperature sensors etc, all with both home assistant and thingsboard integration.

Please check out the wiki, although thats still a work in progress.

Thanks all!

**Comment by andrewleech** [comment]
That looks like a really interesting project, thanks for sharing. I'm always interested in getting more of my home assistant gadgets running on micropython!

**Reply by wolfen351** [reply]
Thanks for looking at it! I'm working on getting more of the wiki filled out. I can't wait for some people to test it out!

**Comment by stephanelsmith** [comment]
Very cool project!  Checking it out!

**Reply by wolfen351** [reply]
Thanks for taking the time to check it out! Would love any feedback you are willing to give.
